### ISSUE TITLE
💅 linter does not detect vue-bound attributes

### ISSUE THREAD

**Issue by rotu** [biomejs/biome#2973]
### Environment information

```bash
$ px biome rage --linter 
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.13.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        correctness/noUndeclaredVariables = "error"
                                correctness/noUnusedVariables = "error"

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedVariables

### Playground link

https://codesandbox.io/p/devbox/staging-resonance-kxcxp4?file=%2Fsrc%2Fmain.vue

### Expected result

The rules `correctness/noUnusedVariables` and `correctness/noUndeclaredVariables` ignore whether variables are used in vue attributes.

```vue
<template>
  <p :style="style1"></p>
  <!-- false negative: /lint/correctness/noUndeclaredVariables -->
  <p :style="style2"></p>

  {{ text }}
</template>
<script setup>
// false positive: /lint/correctness/noUnusedVariables
const style1 = "color:red";
</script>
```

It seems like maybe *only* the `<script>` is checked, not embedded tags, so this might fall under #2239.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by rotu** [comment]
Can this issue please be reopened? It seems like it's now on the roadmap:

> Now that the main architecture is stable and working, in the upcoming patches and minors we will also fix possible inaccuracies and edge cases coming from existing lint rules, such as `noUnusedVariables` inside `<script>` blocks or frontmatter.

https://github.com/biomejs/biome/blob/0d17b05477a537b6d652a2e56c50bb1db013ef06/packages/%40biomejs/biome/CHANGELOG.md?plain=1#L368

**Comment by ematipico** [comment]
No, because it's a known limitation that requires new code, so it's not a bug. It's on our radars. Once we implement it and **we say it's implemented**, we can open issues if it doesn't work

**Comment by rotu** [comment]
That's fair. I guess I'll just have to keep watching the release notes!
